### PR TITLE
Static page docs

### PR DESCRIPTION
**Pull request by johndcollett**
### Description

setting up docs system to handle static pages

### Additional context

---

### Before submitting the PR, please make sure you do the following:

- [ ] Read the [Contributing Guidelines](https://github.com/shopify/hydrogen-ui/blob/main/contributing.md)
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`)
- [ ] Update docs in this repository according to your change
- [ ] Run `yarn changeset add` if this PR cause a version bump based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/). If you have a breaking change, it will need to wait until the next major version release. Otherwise, use patch updates even for new features. Read [more about Hydrogen-UI's versioning.](https://github.com/shopify/hydrogen-ui/blob/main/readme.md#versioning)

### CODE REVIEW

File: packages/react/docs/typeOverride.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
[the remark this answers is not in the document]
Comment (johndcollett) [reply packages/react/docs/typeOverride.json:1]: its just a skeleton file for type overrides so people know where to update it

[the remark this answers is not in the document]
Comment (laurelthorburn) [reply packages/react/docs/typeOverride.json:1]: Nice! 🩻

**Review by laurelthorburn — APPROVED** [review]
Thank you for letting me harass you with questions! Yay!

My final results:
postYarnInstall

**Review by xerilius — APPROVED** [review]
Cool beans 💯

**Pull request merged**